### PR TITLE
manage_app_pool - add periodic recycle control

### PR DESCRIPTION
**Pull request by lcoulson**
manage_app_pool - add periodic recycle control

new param left default undef
logic in class handles undef

caller set to 0 to disable periodic app pool recycling

minutes value converted to ticks and given to powershell for timespan
constructor as int64 max value is powerhsell $([int64]::MaxValue)

validation will reject negatives or out of max bound

### CODE REVIEW

File: manifests/manage_app_pool.pp [hunk 1]
```diff
@@ -16,15 +15,23 @@
   validate_re($start_mode, '^(OnDemand|AlwaysRunning)$')
   validate_bool($rapid_fail_protection)
 
+  if $apppoolrecycleperiodicminutes != undef {
+    validate_integer($apppoolrecycleperiodicminutes, 15372286728, 0) # powershell $([int64]::MaxValue) / 600000000, we're not dealing with negative
```
Comment (juniorsysadmin) [inline manifests/manage_app_pool.pp]: You probably should make these numbers constants, variables or parameters, defined here or elsewhere

File: manifests/manage_app_pool.pp [hunk 2]
```diff
@@ -16,15 +15,23 @@
...
+    $periodicticks        = $apppoolrecycleperiodicminutes * 600000000
```
Comment (juniorsysadmin) [inline manifests/manage_app_pool.pp]: As above

**Comment by lcoulson** [comment]
@igalic @juniorsysadmin - added tests, standardized variable naming, added documentation example, updated stdlib min version to allow for validate_integer()

**Comment by juniorsysadmin** [comment]
Looks okay to me

**Pull request merged**